### PR TITLE
# Add analsis URL

### PR DESCRIPTION
**Pull request by pockerman**
This is the first round for actually having the GUI to compute sth...so let's discuss

### CODE REVIEW

**Comment by pockerman** [comment]
Awesome you added Travis CI also brilliant

File: pom.xml [hunk 1]
```diff
@@ -3,8 +3,7 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
-
-    <packaging>war</packaging>
+    
     <groupId>jstat.gui</groupId>
     <artifactId>jstatgui</artifactId>
     <version>0.0.1</version>
```
Comment (jethronap) [inline pom.xml]: pls use the pom file we have in the master branch.

File: src/main/java/detail/datasets/IDataSetContainer.java [hunk 2]
```diff
@@ -23,4 +26,15 @@
...
+
+    /**
+     * Returns  a list of the loaded dataset names
+     */
+    List<String> dataSetNames();
+
+    /**
+     * Clear whatever the data set  may contain
+     */
+    void clear();
```
Comment (jethronap) [inline src/main/java/detail/datasets/IDataSetContainer.java]: why do we need a clear function here? maybe it's too risky to give the user the option to delete the contents of the data set.

File: src/main/java/detail/tasks/ComputeDescriptiveStatisticsTask.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import java.util.List;
+import java.util.concurrent.Callable;
+
+/**
+ * Compute the descriptive statistics of the columns from the given
+ * data set
+ */
+public class ComputeDescriptiveStatisticsTask extends TaskBase {
+
+    public ComputeDescriptiveStatisticsTask(String name, IDataSet dataSet, String... names){
```
Comment (jethronap) [inline src/main/java/detail/tasks/ComputeDescriptiveStatisticsTask.java]: what do the ... mean?

File: src/main/java/gui/Analysis.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        }
+        else{
+
+            // submit it to the pool
+            TaskBase task = new ComputeDescriptiveStatisticsTask("Mean", dataSet, formWrapper.colName);
+            JStatGuiGlobalData.workersPool.submit(task);
+            JStatGuiGlobalData.tasks.add(task);
+        }
+    }
+}
```
Comment (jethronap) [inline src/main/java/gui/Analysis.java]: as I told you we need to revisit the controllers as far as the values are concerned.

File: src/main/java/gui/AnalysisResult.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        else{
+
+            // return an error message that no valid action
+            // was selected
+        }*/
+
+        // redirect to the analysis page again
+        return "redirect:/analysis-result?taskName="+taskName;
+    }
+}
```
Comment (jethronap) [inline src/main/java/gui/AnalysisResult.java]: same as above

File: src/main/resources/templates/analysis_index.html [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+
+<div>
+
+</div>
+    <button>Do Analysis</button>
+</form>
+
+</body>
+</html>
```
Comment (jethronap) [inline src/main/resources/templates/analysis_index.html]: we can use the frontend files from the redefine_beautify branch, I think. 
but they are ok for now.

File: pom.xml (hunk 1 above)
Comment (pockerman) [reply pom.xml]: yes I should have already done that

File: src/main/java/detail/datasets/IDataSetContainer.java (hunk 2 above)
Comment (pockerman) [reply src/main/java/detail/datasets/IDataSetContainer.java]: This class is meant to fold all the datasets loaded by the app. Once the user clicks on Quit everything should be dumped. This is just a collection of data sets and not an actual data set

File: src/main/java/detail/tasks/ComputeDescriptiveStatisticsTask.java (hunk 3 above)
Comment (pockerman) [reply src/main/java/detail/tasks/ComputeDescriptiveStatisticsTask.java]: yes I need to comment this. Wrote it yesterday on the fly. I need sth to distinguish that tasks submitted to the work pool so ```name``` is the task name which should be globally unique.
```dataSet``` is the data set instance for which the specific computation was requetsted. The
```ComputeDescriptiveStatisticsTask``` is meant to compute descriptive statistics for the columns specified in the list ```names``` of the given ```dataSet``` instance

File: src/main/java/gui/Analysis.java (hunk 4 above)
Comment (pockerman) [reply src/main/java/gui/Analysis.java]: Yes i agree but to my mind the controllers are just to serve requests this is not a request you just pust a computation specified by the ```task```. In fact this is not work pool is just an extra array to query in O(N) time but still need to figure out how are we going to clean this from the tasks that have been executed :) :)

File: src/main/resources/templates/analysis_index.html (hunk 6 above)
Comment (pockerman) [reply src/main/resources/templates/analysis_index.html]: ok

File: src/main/java/gui/Analysis.java (hunk 4 above)
Comment (pockerman) [reply src/main/java/gui/Analysis.java]: In any case I think we need an ```Analysis``` controller to start any kind of analysis and then redirect to the analogous result page as in  ```"redirect:/analysis-result?taskName="+taskName;```

**Pull request merged**